tower-buffer/src/lib.rs

```diff
@@ -27,16 +27,16 @@ use std::sync::atomic::Ordering;
 /// Adds a buffer in front of an inner service.
 ///
 /// See crate level documentation for more details.
-pub struct Buffer<T>
-where T: Service,
+pub struct Buffer<T, R>
+where T: Service<R>,
 {
-    tx: UnboundedSender<Message<T>>,
+    tx: UnboundedSender<Message<T, R>>,
     state: Arc<State>,
 }
 
 /// Future eventually completed with the response to the original request.
-pub struct ResponseFuture<T>
-where T: Service,
+pub struct ResponseFuture<T, R>
+where T: Service<R>,
```

Comment:
Additional generic type parameters were added to most of the types in `tower-buffer` for the request types of the wrapped `Service`, rather than removing the `T: Service` bounds. This was necessary since the associated types from the service type parameter are also used in struct definitions, so it was necessary to keep the `T: Service` bound